### PR TITLE
[DEM-266] Correct titlecase regex bug in `FeatureFlagNameValid` cop

### PR DESCRIPTION
**Pull request by j-sm-n**
## What did we change?
Fix a bug in `FeatureFlagNameValid` cop.

## Why are we doing this?
There was a typo in the `TITLECASE_SEGMENT` regex, causing it to incorrectly identify valid flag names.

## How was it tested?
- [x] Specs
- [ ] Locally

### CODE REVIEW

**Review by tristandunn — APPROVED** [review]

**Comment by j-sm-n** [comment]
I've reached out to IT to see if they can help with the CircleCI error.

**Comment by jmpage** [comment]
@j-sm-n I fixed the blocker by adding a read-only deploy key for you (from Circle CI project settings). It looks like the build is now failing on rubocop.

**Review by mikowitz — COMMENTED** [review]
Looks good to me as soon as the CI issues are cleaned up!

File: .circleci/config.yml [hunk 1]
```diff
@@ -39,4 +39,4 @@ workflows:
           matrix:
             parameters:
               # https://github.com/CircleCI-Public/cimg-ruby
-              ruby-version: ["2.6", "2.7", "3.0", "3.1"]
```
Comment (j-sm-n) [inline .circleci/config.yml:42]: Since Ruby 2.6 has been EOL for about a year now, it made sense to me to remove CI support for it.

**Review by tristandunn — APPROVED** [review]
💯

**Pull request merged**